src/service/grpcweb.ts

```diff
@@ -221,22 +239,217 @@ function generateJavaScript(fileDescriptor: FileDescriptorProto, exportMap: Expo
+        .indent().printLn(`listeners.data.forEach(function (callback) {`)
+          .indent().printLn(`callback(responseMessage);`)
+        .dedent().printLn(`});`)
+      .dedent().printLn(`},`)
+               .printLn(`onEnd: function () {`)
+        .indent().printLn(`listeners.end.forEach(function (callback) {`)
+          .indent().printLn(`callback();`)
+        .dedent().printLn(`});`)
+                 .printLn(`listeners.data = [];`)
+                 .printLn(`listeners.end = [];`)
+      .dedent().printLn(`}`)
+    .dedent().printLn(`});`)
+             .printLn(`return {`)
+      .indent().printLn(`on: function (eventType, callback) {`)
+        .indent().printLn(`listeners[eventType] = callback;`)
+      .dedent().printLn(`}`)
+    .dedent().printLn(`};`)
+  .dedent().printLn(`};`);
+}
+
+function printBidirectionalStubMethod(
+  printer: CodePrinter,
+  service: RPCDescriptor,
+  camelCaseMethodName: string
+) {
+  printer
+           .printLn(`${service.name}.prototype.${camelCaseMethodName} = function ${camelCaseMethodName}() {`)
+    .indent().printLn(`throw new Error("Client streaming is not currently supported");`)
+  .dedent().printLn(`}`);
+}
+function printClientStreamStubMethod(
+  printer: CodePrinter,
+  service: RPCDescriptor,
+  camelCaseMethodName: string
+) {
+  printer
+           .printLn(`${service.name}.prototype.${camelCaseMethodName} = function ${camelCaseMethodName}() {`)
+    .indent().printLn(`throw new Error("Bi-directional streaming is not currently supported");`)
+  .dedent().printLn(`}`);
+}
+
+function printServiceStubTypes(methodPrinter: Printer, service: RPCDescriptor) {
+  const printer = new CodePrinter(0, methodPrinter);
+
+  printer
+           .printLn(`export class ${service.name}Client {`)
+    .indent().printLn(`serviceHost: string;`)
+             .printLn(`constructor(serviceHost: string);`)
+
+  service.methods.forEach((method: RPCMethodDescriptor) => {
+
+    const camelCaseMethodName = method.name[0].toLowerCase() + method.name.substr(1);
+
+    if (method.requestStream && method.responseStream) {
+      printBidirectionalStubMethodTypes(printer, camelCaseMethodName);
+    } else if (method.requestStream) {
+      printClientStreamStubMethodTypes(printer, camelCaseMethodName);
+    } else if (method.responseStream) {
+      printServerStreamStubMethodTypes(printer, method, camelCaseMethodName);
+    } else {
+      printUnaryStubMethodTypes(printer, method, camelCaseMethodName);
+    }
+  });
+  printer.dedent().printLn("}");
+}
+
+function printUnaryStubMethodTypes(
+  printer: CodePrinter,
+  method: RPCMethodDescriptor,
+  camelCaseMethodName: string
+) {
+  printer
+             .printLn(`${camelCaseMethodName}(`)
+      .indent().printLn(`requestMessage: ${method.requestType},`)
+               .printLn(`metadata?: grpc.Metadata,`)
+               .printLn(`callback?: (error: any, responseMessage: ${method.responseType}|null) => void`)
```

Comment:
can the error argument be typed as `Error` instead of `any`?

Reply:
Unless you know for sure that it's always an actual Error object, I'd be averse to set that type.